gcp_variant_transforms/transforms/infer_undefined_headers.py

```diff
@@ -64,9 +62,21 @@ def _get_field_type(self, field_value):
       return vcf_field_conflict_resolver.VcfParserConstants.INTEGER
     elif isinstance(field_value, float):
       return vcf_field_conflict_resolver.VcfParserConstants.FLOAT
+    elif self._can_cast_to(field_value, int):
```

Comment:
Consider doing `elif isinstance(field_value, int) or self._can_cast_to(int)` above.

Reply:
It would be wrong to do so, and you will end up inferring int type for float values (and vice versa) because you can always cast float to int.

Reply:
ah! nice catch! Just making sure: is this already covered by unit tests? I just want to avoid someone else making the same mistake if they end up refactoring this code in the future. Perhaps you can also add a comment about it.

Reply:
Good point. We have a test for int -> float. Added one for making sure float -> int doesn't happen.

Reply:
Perfect. Thanks!